axelor-sale/src/main/java/com/axelor/apps/sale/service/ProductRestServiceImpl.java

```diff
@@ -0,0 +1,122 @@
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.base.db.Product;
+import com.axelor.apps.base.db.repo.PartnerRepository;
+import com.axelor.apps.base.db.repo.TraceBackRepository;
+import com.axelor.apps.base.service.CompanyService;
+import com.axelor.apps.base.service.ProductPriceService;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.apps.base.service.tax.TaxService;
+import com.axelor.apps.base.service.user.UserService;
+import com.axelor.apps.sale.exception.SaleExceptionMessage;
+import com.axelor.apps.sale.rest.dto.CurrencyResponse;
+import com.axelor.apps.sale.rest.dto.PriceResponse;
+import com.axelor.apps.sale.rest.dto.ProductResponse;
+import com.axelor.apps.sale.service.app.AppSaleService;
+import com.axelor.auth.AuthUtils;
+import com.axelor.auth.db.User;
+import com.axelor.i18n.I18n;
+import com.google.inject.Inject;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ProductRestServiceImpl implements ProductRestService {
+
+  protected AppSaleService appSaleService;
+  protected CompanyService companyService;
+
+  protected PartnerRepository partnerRepository;
+  protected TaxService taxService;
+  protected UserService userService;
+  protected ProductRestService productRestService;
+  protected AppBaseService appBaseService;
+  protected ProductPriceService productPriceService;
+
+  @Inject
+  public ProductRestServiceImpl(
+      AppSaleService appSaleService,
+      CompanyService companyService,
+      PartnerRepository partnerRepository,
+      UserService userService,
+      ProductRestService productRestService,
+      AppBaseService appBaseService,
+      ProductPriceService productPriceService) {
+    this.appSaleService = appSaleService;
+    this.companyService = companyService;
+    this.userService = userService;
+    this.partnerRepository = partnerRepository;
+    this.productRestService = productRestService;
+    this.appBaseService = appBaseService;
+    this.productPriceService = productPriceService;
+  }
+
+  @Override
+  public List<PriceResponse> fetchProductPrice(Product product, Partner partner, Company company)
+      throws AxelorException {
+    checkProduct(product);
+    List<PriceResponse> priceList = new ArrayList<>();
+
+    BigDecimal priceWT = productPriceService.getSaleUnitPrice(company, product, false, partner);
+    BigDecimal priceATI = productPriceService.getSaleUnitPrice(company, product, true, partner);
+
+    priceList.add(new PriceResponse("WT", priceWT));
+    priceList.add(new PriceResponse("ATI", priceATI));
+    return priceList;
+  }
+
+  protected void checkProduct(Product product) throws AxelorException {
+    if (product == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_NO_VALUE, I18n.get(SaleExceptionMessage.PRODUCT_IS_NULL));
+    }
+  }
```

Comment:
I don't think the product can be null here. if we send an id which does not exist, we should get an exception when finding the product in the rest controller. If we do not send an id, we should get an exception from RequestValidator.